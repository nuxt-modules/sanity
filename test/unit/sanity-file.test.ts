import { expect, describe, it } from 'vitest'
import { mount } from '@vue/test-utils'
import { h } from 'vue'

import SanityFile from '../../src/runtime/components/sanity-file'

describe('SanityFile', () => {
  it('provides a valid renderless component', () => {
    const wrapper = mount(SanityFile,
      {
        props: {
          assetId: 'file-41773b5c55bc5414ab7554a75eefddf8e2e14524-txt',
        },
        slots: {
          default: ({ src }) => h('a', { href: src }, 'Click here to read'),
        },
      },
    )

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('allows injecting download params', () => {
    const wrapper = mount(SanityFile,
      {
        props: {
          assetId: 'file-41773b5c55bc5414ab7554a75eefddf8e2e14524-txt',
          download: 'myfile.txt',
        },
        slots: {
          default: ({ src }) => h('a', { href: src }, 'Click here to download'),
        },
      },
    )

    expect(wrapper.html()).toMatchSnapshot()
  })
})
