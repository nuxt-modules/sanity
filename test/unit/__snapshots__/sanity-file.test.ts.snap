// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SanityFile > allows injecting download params 1`] = `"<a href=\\"https://cdn.sanity.io/files/test-project/production/41773b5c55bc5414ab7554a75eefddf8e2e14524.txt?dl=myfile.txt\\">Click here to download</a>"`;

exports[`SanityFile > provides a valid renderless component 1`] = `"<a href=\\"https://cdn.sanity.io/files/test-project/production/41773b5c55bc5414ab7554a75eefddf8e2e14524.txt\\">Click here to read</a>"`;
